src/IO/WriteHelpers.cpp

```diff
@@ -52,7 +52,7 @@ void writeException(const Exception & e, WriteBuffer & buf, bool with_stack_trac
 {
     writeBinary(e.code(), buf);
     writeBinary(String(e.name()), buf);
-    writeBinary(e.displayText(), buf);
+    writeBinary(e.displayText() + getExtraExceptionInfo(e), buf);
```

Comment:
No, it writes string size:
https://github.com/ClickHouse/ClickHouse/blob/a1e54c39185298a4cc3a14d3381f10ab8871db54/src/IO/WriteHelpers.h#L99-L103

Reply:
But it will be backward incompatible. Also "extra exception info" is actually a part of exception message. The only thing we can do with separate "extra info" after reading it is to concatenate it with "display text". That's what `getCurrentExceptionMessage` do:
https://github.com/ClickHouse/ClickHouse/blob/94af11dd74a45fe43fda61ce5421537d25a19ee2/src/Common/Exception.cpp#L312-L324
The problem is that additional diagnostic info was not added when sending exception producing exception messages like this:
```
Received exception from server (version 22.5.1):
Code: 76. DB::Exception: Received from bxkmiu60ki.eu-west-1.aws.clickhouse.cloud:9440. DB::ErrnoException. DB::ErrnoException: Cannot open file /var/lib/clickhouse/disks/s3disk/store/8d4/8d482cf0-4a61-47af-a09e-42d9420e10fc/tmp_insert_all_314150_314150_0/path.mrk2, errno: 28, strerror: No space left on device. (CANNOT_OPEN_FILE)

```
(without information about total/free space and inodes)

Reply:
We could solve the problem with backward incompatibility by introducing a new `DBMS_MIN_PROTOCOL_VERSION_`, but probably this task doesn't worth that.